---
title: "Clonal Evolution Figure 1"
author: "Robert Bowman"
date: "5/12/2022"
output: html_document
---

Setup and Packages
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = "/Volumes/LevineLab/Levine Lab/Bobby/Project Notes/Flt3/")
```

Data Sets
```{r, message=FALSE}
library(tidyverse)
library(cowplot)
library(RColorBrewer)
library(pals)
setwd("/Volumes/LevineLab/Levine Lab/Bobby/Project Notes/Flt3/")
source("./Scripts/extra_functions.R")
results_Leuk<-readRDS(file="./Results/2022/February/Flt3_Npm1_3DE_STAR_results.rds")
pheno_select_leuk<-readRDS(file="./Results/2022/February/Flt3_NPM1_3DE_STAR_pheno.rds")
norm_counts_leuk<-readRDS(file="./Results/2022/February/Flt3_NPM1_3DE_STAR_norm_counts.rds")
long_counts_leuk<-readRDS(file="./Results/2022/February/Flt3_NPM1_3DE_STAR_long_counts.rds")
vsd_transformation_leuk<-readRDS(file="./Results/2022/February/Flt3_NPM1_3DE_STAR_vsd.rds")

results_FTime<-readRDS(file="./Results/2022/February/Flt3_3DE_STAR_results.rds")
pheno_FTime<-readRDS(file="./Results/2022/February/Flt3_3DE_STAR_pheno.rds")
norm_counts_FTime<-readRDS(file="./Results/2022/February/Flt3_3DE_STAR_norm_counts.rds")
long_counts_FTime<-readRDS(file="./Results/2022/February/Flt3_3DE_STAR_long_counts.rds")
vsd_transformation_FTime<-readRDS(file="./Results/2022/February/Flt3_3DE_STAR_vsd.rds")
```

Gene lists
```{r}
immediate_response_GMP<-read.csv(file="./Results/2022/February/immediate_response_GMP_genes.csv")[,-1]
sustained_response_GMP<-read.csv(file="./Results/2022/February/sustained_response_GMP_genes.csv")[,-1]
delayed_response_GMP<-read.csv(file="./Results/2022/February/delayed_response_GMP_genes.csv")[,-1]

immediate_response_LSK<-read.csv(file="./Results/2022/February/immediate_response_LSK_genes.csv")[,-1]
sustained_response_LSK<-read.csv(file="./Results/2022/February/sustained_response_LSK_genes.csv")[,-1]
delayed_response_LSK<-read.csv(file="./Results/2022/February/delayed_response_LSK_genes.csv")[,-1]
```

GSEA setup
```{R gene set extraction and setup,cache=TRUE}
library(msigdbr)
library(fgsea)
C2_gene_sets <- msigdbr(species = "Mus musculus", category = "C2")%>%
                          data.frame%>%
                          dplyr::select(gene_symbol,gs_name)%>%
                          split(f = as.factor(.$gs_name))%>%
                          lapply(function(x){x$gene_symbol})
H_gene_sets <- msigdbr(species = "Mus musculus", category = "H")%>%
                          data.frame%>%
                          dplyr::select(gene_symbol,gs_name)%>%
                          mutate(gs_name=gsub("HALLMARK_","",gs_name))%>%
                          split(f = as.factor(.$gs_name))%>%
                          lapply(function(x){x$gene_symbol})


```

Figure 1E
```{r}
library(tidyheatmaps)
genes_of_interest<- c("Gata1","Gata2","Socs1","Cish","Socs2",
                        "Itgax","Egr1","Junb","Fosb",'Jund',
                      "Tnf","Irf7","Irf8")
long_counts_FTime%>%
  dplyr::filter(Gene%in%c(genes_of_interest))%>%
  dplyr::arrange(Cell,desc(Genotype), Timepoint)%>%
  dplyr::mutate(value=scale(log2(value+0.1)))%>%
  dplyr::filter(Cell=="LSK")%>%
  tidy_heatmap(rows=Gene,
               columns = Sample,
               values = value,
               scale= "row",
               annotation_col =c(Cell,Genotype,Timepoint),
               cluster_rows=T,
               colors = rev(brewer.rdbu(n=100)),
               color_legend_n=100,
              # gaps_row = gene_set,
               treeheight_row = 0,
               cellwidth = 5,
               cellheight = 7,
               show_colnames = FALSE,
                legend = TRUE,

               annotation_colors = list("Genotype"=c("WT"="grey70",
                                                     "Flt3"=brewer.pal(n=5,"Reds")[5]),
                                        "Cell"=c("LSK"=brewer.pal(n=5,"Purples")[5],
                                                 "GMP"=brewer.pal(n=5,"Oranges")[5]),
                                          "Timepoint"=c("Control"=brewer.pal(n=5,"Greys")[1],
                                                      "2w"=brewer.pal(n=5,"Greys")[2],
                                                      "4w"=brewer.pal(n=5,"Greys")[3],
                                                      "6w"=brewer.pal(n=5,"Greys")[4],
                                                      "8w"=brewer.pal(n=5,"Greys")[5])) ,
               file="./Mature_analysis/Final_Figures/Figure 1/Figure1e-heatmap.pdf")

```

Figure 1F
```{r}
library(DESeq2)
library(GSVA)
library(pals)
LSK_samples<-pheno_FTime%>%filter(Cell=="LSK")%>%pull(Sample)
vsd_transformation_FTime_LSK<-vsd_transformation_FTime[,LSK_samples]

vsd_FTime_scale<- (assay(vsd_transformation_FTime_LSK)-rowMeans(assay(vsd_transformation_FTime_LSK)))/rowSds(assay(vsd_transformation_FTime_LSK))
gsva_results_FTime<-gsva(vsd_FTime_scale,
                          H_gene_sets,
                          min.sz=10, max.sz=5000,
                          method="ssgsea",
                         # rnaseq=TRUE, 
                          verbose=TRUE)%>%
                      data.frame()%>%
                      rownames_to_column(var="Gene_set")%>%
                      pivot_longer(cols=starts_with("IL"),names_to="Sample",values_to="value")%>%
                      left_join(pheno_FTime,by="Sample")%>%
                      unite(col=Composite,Genotype,Timepoint,sep = "_",remove = F)%>%
                      mutate(Composite=factor(Composite,levels=c("WT_4w","Flt3_2w","Flt3_4w","Flt3_6w","Flt3_8w")))
                    #  mutate(Genotype=factor(Genotype,levels=c("Flt3","WT")))

genes_of_interest<- c(#"IL6_JAK_STAT3_SIGNALING",
                  #    "E2F_TARGETS",
                      "HEME_METABOLISM",
                  #   "KRAS_SIGNALING_DN",
                #   "KRAS_SIGNALING_UP",
                     "IL2_STAT5_SIGNALING",
                   #   "TNFA_SIGNALING_VIA_NFKB",
                      "MYC_TARGETS_V2")

ggssGSEA_LSK<-ggplot(gsva_results_FTime%>%
         # filter(Cell=="LSK")%>%
          filter(Gene_set%in%genes_of_interest)%>%
          mutate(Gene_set=factor(Gene_set)),
          aes(x=Composite, y=value,group=Gene_set,
           color=Gene_set))+
  stat_summary(fun.data = mean_se, width=0.25 , geom = "errorbar") + 
  stat_summary(fun=mean, geom="line", aes(group = Gene_set)) + 
  #facet_grid(.~Cell,scale="free")+
  ylab("Relative enrichment \n (ssGSEA score)")+
  theme_classic(base_size=8)+
  xlab("Group")+
  scale_color_manual(values=kelly(n=length(genes_of_interest)+1)[-1])+
 # guides(color="none")+
  #scale_y_continuous(limits=c(-0.2,1))+
   theme(strip.text.y=element_text(angle=0),
        axis.title.x = element_blank(),
        plot.title = element_blank(),
      #  legend.position = "right",
       # legend.title = element_blank(),
        axis.text.x=element_text(angle=30,hjust=1,vjust=1),
        strip.background = element_blank())

ggsave(ggssGSEA_LSK,#,textSize = 5),
       width=2,
       height=2,
       file="./Mature_analysis/Final_Figures/Figure 1/Figure1f-ssGSEA.pdf")
```
