---
title: "Clonal Evolution Figure 1"
author: "Robert Bowman"
date: "5/4/2022"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = "/Users/bowmanrl/Projects/Flt3_GL/Server_dump/FLT3_GL")
```

```{r, message=FALSE}
library(tidyverse)
library(cowplot)
library(RColorBrewer)
library(pals)
setwd("/Users/bowmanrl/Projects/Flt3_GL/Server_dump/FLT3_GL")
#source("./Scripts/extra_functions.R")
results_Leuk<-readRDS(file="./Results/2022/February/Flt3_Npm1_3DE_STAR_results.rds")
pheno_select_leuk<-readRDS(file="./Results/2022/February/Flt3_NPM1_3DE_STAR_pheno.rds")
norm_counts_leuk<-readRDS(file="./Results/2022/February/Flt3_NPM1_3DE_STAR_norm_counts.rds")
long_counts_leuk<-readRDS(file="./Results/2022/February/Flt3_NPM1_3DE_STAR_long_counts.rds")
vsd_transformation_leuk<-readRDS(file="./Results/2022/February/Flt3_NPM1_3DE_STAR_vsd.rds")

results_FTime<-readRDS(file="./Results/2022/February/Flt3_3DE_STAR_results.rds")
pheno_FTime<-readRDS(file="./Results/2022/February/Flt3_3DE_STAR_pheno.rds")
norm_counts_FTime<-readRDS(file="./Results/2022/February/Flt3_3DE_STAR_norm_counts.rds")
long_counts_FTime<-readRDS(file="./Results/2022/February/Flt3_3DE_STAR_long_counts.rds")
vsd_transformation_FTime<-readRDS(file="./Results/2022/February/Flt3_3DE_STAR_vsd.rds")
```

Gene lists
```{r}
immediate_response_GMP<-read.csv(file="./Results/2022/February/immediate_response_GMP_genes.csv")[,-1]
sustained_response_GMP<-read.csv(file="./Results/2022/February/sustained_response_GMP_genes.csv")[,-1]
delayed_response_GMP<-read.csv(file="./Results/2022/February/delayed_response_GMP_genes.csv")[,-1]

immediate_response_LSK<-read.csv(file="./Results/2022/February/immediate_response_LSK_genes.csv")[,-1]
sustained_response_LSK<-read.csv(file="./Results/2022/February/sustained_response_LSK_genes.csv")[,-1]
delayed_response_LSK<-read.csv(file="./Results/2022/February/delayed_response_LSK_genes.csv")[,-1]
cutoff<- -log10(0.01)
immediate_LSK_up<-immediate_response_LSK%>%dplyr::filter(Flt3_2w_LSK>cutoff)%>%pull(Genes)
sustained_LSK_up<-sustained_response_LSK%>%dplyr::filter(Flt3_8w_LSK>cutoff)%>%pull(Genes)
delayed_LSK_up<-delayed_response_LSK%>%dplyr::filter(Flt3_4w_LSK>cutoff)%>%pull(Genes)

immediate_GMP_up<-immediate_response_GMP%>%dplyr::filter(Flt3_2w_GMP>cutoff)%>%pull(Genes)
sustained_GMP_up<-sustained_response_GMP%>%dplyr::filter(Flt3_8w_GMP>cutoff)%>%pull(Genes)
delayed_GMP_up<-delayed_response_GMP%>%dplyr::filter(Flt3_4w_GMP>cutoff)%>%pull(Genes)

immediate_LSK_down<-immediate_response_LSK%>%dplyr::filter(Flt3_2w_LSK<(-cutoff))%>%pull(Genes)
sustained_LSK_down<-sustained_response_LSK%>%dplyr::filter(Flt3_8w_LSK<(-cutoff))%>%pull(Genes)
delayed_LSK_down<-delayed_response_LSK%>%dplyr::filter(Flt3_4w_LSK<(-cutoff))%>%pull(Genes)

immediate_GMP_down<-immediate_response_GMP%>%dplyr::filter(Flt3_2w_GMP<(-cutoff))%>%pull(Genes)
sustained_GMP_down<-sustained_response_GMP%>%dplyr::filter(Flt3_8w_GMP<(-cutoff))%>%pull(Genes)
delayed_GMP_down<-delayed_response_GMP%>%dplyr::filter(Flt3_4w_GMP<(-cutoff))%>%pull(Genes)

```

Extended Data 2A
```{r}
PCA<-plotPCA(vsd_transformation_FTime,
                            intgroup="Composite",
                            ntop=1000,
                            return=TRUE)

ggPCA<-ggplot(PCA%>%
                inner_join(pheno_FTime%>%dplyr::select(Sample,Cell),by=c("name"="Sample")),
               aes(x=PC1,y=PC2,color=Composite,shape=Cell))+
  geom_point(size=3)+
  theme_minimal(base_size = 8)+
  theme(axis.line.x = element_line(),
        axis.line.y = element_line())+
  scale_color_manual(values=c(pals::kelly()[-1]))

ggsave(addSmallLegend(ggPCA,textSize = 8),
      width=3,
      height=2.25,
  file="./Mature_analysis/Final_Figures/Extended Data 2/ExData_2a-PCA.pdf")
```

Extended Data 2B,C
```{r message=FALSE}

gg_LSK_timecourse_input<-long_counts_FTime%>%
    dplyr::filter(Cell=="LSK")%>%
    group_by(Gene,Composite)%>%
    summarize(value_mean=log2(mean(value+1)))%>%
    ungroup()%>%
    group_by(Gene)%>%
    mutate(value_mean_fc=value_mean-value_mean[Composite=="WT_4w"])  %>%
    ungroup()%>%
    mutate(Gene_group=ifelse(Gene%in%immediate_LSK_up,"Immediate Up",
                      ifelse(Gene%in%sustained_LSK_up,"Sustained Up", 
                      ifelse(Gene%in%delayed_LSK_up,"Delayed Up",
                                    ifelse(Gene%in%immediate_LSK_down,"Immediate Down",
                                    ifelse(Gene%in%sustained_LSK_down,"Sustained Down", 
                                    ifelse(Gene%in%delayed_LSK_down,"Delayed Down", 
                                           "Other")))))))%>%
    mutate(Gene_group=factor(Gene_group,levels=c("Sustained Up","Delayed Up","Immediate Up",
                                                 "Sustained Down","Delayed Down","Immediate Down")))%>%
    mutate(Comp_continuous=ifelse(Composite=="WT_4w",0,
                                 ifelse(Composite=="Flt3_2w",2, 
                                        ifelse(Composite=="Flt3_4w",4, 
                                               ifelse(Composite=="Flt3_6w",6, 
                                                      ifelse(Composite=="Flt3_8w",8, NA))))))%>%
    dplyr::filter(Gene_group!="Other")


gg_LSK_timecourse<-gg_LSK_timecourse_input%>%
                  ggplot(aes(x=Comp_continuous,y=value_mean_fc,color=Gene_group,group=Gene))+
                  rasterize(geom_line(alpha=0.05,lwd=0.25))+
                  stat_summary(fun=mean,lty=5,lwd=0.75, geom="line",aes(group = Gene_group))+
                  scale_y_continuous(limits = c(-7.5,7.5))+
                  scale_color_manual(values=c("Sustained Up"=stepped()[9],
                                              "Delayed Up"=stepped()[17],
                                              "Immediate Up"=stepped()[1],
                                              "Sustained Down"=stepped()[13],
                                              "Delayed Down"=stepped()[5],
                                              "Immediate Down"=stepped3()[5]))+
                    ylab("Log2 Fold Change from WT")+
                    xlab("Weeks post TAM")+
                              theme_classic(base_size=8)

gg_timecourse_barplot<-gg_LSK_timecourse_input%>%
                        distinct(Gene,Gene_group)%>%
                  ggplot(aes(y=Gene_group,fill=Gene_group))+
                  geom_bar(stat="count")+
                  scale_y_discrete(limits = rev(levels(all_of(gg_LSK_timecourse_input$Gene_group))),
                                   breaks= rev(levels(all_of(gg_LSK_timecourse_input$Gene_group))),
                                   labels= rev(c("Sustained \n Up",
                                             "Delayed \n Up",
                                             "Immediate \n Up",
                                            "Sustained \n Down",
                                             "Delayed \n Down",
                                             "Immediate \n Down" )))+
                  scale_fill_manual(values=c("Sustained Up"=stepped()[9],
                                              "Delayed Up"=stepped()[17],
                                              "Immediate Up"=stepped()[1],
                                              "Sustained Down"=stepped()[13],
                                              "Delayed Down"=stepped()[5],
                                              "Immediate Down"=stepped3()[5]))+
                  guides(fill="none")+
                  ylab("Gene Group")+
                  xlab("# of Genes")+
                              theme_classic(base_size=8)

ggsave(
  plot_grid(gg_timecourse_barplot,addSmallLegend(gg_LSK_timecourse,textSize = 8),rel_widths = c(0.6,1)),
  width=5,height=2.25,
  file="./Mature_analysis/Final_Figures/Extended Data 2/ExData_2bc.pdf")
```

Extended Data 2D,F
```{r}
genes_of_interest<- c("Tal1","Meis1","Mecom","Hoxa9","Hoxb5")

ggstem_FTime<-ggplot(long_counts_FTime%>%
          filter(Gene%in%genes_of_interest)%>%
          filter(Cell=="LSK")%>%
          mutate(Gene=factor(Gene,levels=genes_of_interest)),
          aes(x=factor(interaction(Genotype,Timepoint),
                       levels=c("WT.4w","Flt3.2w","Flt3.4w","Flt3.6w","Flt3.8w")),
             fill=factor(interaction(Genotype,Timepoint),
                       levels=c("WT.4w","Flt3.2w","Flt3.4w","Flt3.6w","Flt3.8w")),
            y=value))+
            stat_summary(fun= median, geom = "crossbar", width = 0.25,color="black",lwd=0.25) + 
            stat_summary(fun.data = mean_se, geom = "errorbar", width = 0.25,color="black",lwd=0.25)+
            geom_point(size=1.5,pch=21,lwd=0.5)+
         facet_grid(Gene~Cell,scale="free")+
              ylab("Normalized expression")+
              xlab("Group")+
          scale_fill_manual(values=c("WT.4w"=brewer.pal(5,"Greys")[3],
                                     "Flt3.2w"=brewer.pal(5,"Reds")[1],
                                      "Flt3.4w"=brewer.pal(5,"Reds")[2],
                                      "Flt3.6w"=brewer.pal(5,"Reds")[3],
                                      "Flt3.8w"=brewer.pal(5,"Reds")[5]),"Group")+
               guides(fill="none")+
          theme_classic(base_size=8)+
              geom_hline(aes(yintercept=-Inf)) + 
              geom_vline(aes(xintercept=-Inf)) + 
              coord_cartesian(clip="off")+
         #     coord_flip()+
        #      scale_x_discrete(limits=rev(c("WT.4w","Flt3.2w","Flt3.4w","Flt3.6w","Flt3.8w")))+
              theme(strip.text.y=element_text(angle=0),
                    strip.background = element_blank(),
                    axis.title.x = element_blank(),
                    axis.text.x=element_text(angle=30,hjust=1,vjust=1))


genes_of_interest<- c("Socs2","Pim2","Spp1","Mcl1","Bcl2")

ggSTAT_FTime<-ggplot(long_counts_FTime%>%
          filter(Gene%in%genes_of_interest)%>%
          filter(Cell=="LSK")%>%
          mutate(Gene=factor(Gene,levels=genes_of_interest)),
          aes(x=factor(interaction(Genotype,Timepoint),
                       levels=c("WT.4w","Flt3.2w","Flt3.4w","Flt3.6w","Flt3.8w")),
             fill=factor(interaction(Genotype,Timepoint),
                       levels=c("WT.4w","Flt3.2w","Flt3.4w","Flt3.6w","Flt3.8w")),
            y=value))+
            stat_summary(fun= median, geom = "crossbar", width = 0.25,color="black",lwd=0.25) + 
            stat_summary(fun.data = mean_se, geom = "errorbar", width = 0.25,color="black",lwd=0.25)+
            geom_point(size=1.5,pch=21,lwd=0.5)+
         facet_grid(Gene~Cell,scale="free")+
              ylab("Normalized expression")+
              xlab("Group")+
          scale_fill_manual(values=c("WT.4w"=brewer.pal(5,"Greys")[3],
                                     "Flt3.2w"=brewer.pal(5,"Reds")[1],
                                      "Flt3.4w"=brewer.pal(5,"Reds")[2],
                                      "Flt3.6w"=brewer.pal(5,"Reds")[3],
                                      "Flt3.8w"=brewer.pal(5,"Reds")[5]),"Group")+
               guides(fill="none")+
          theme_classic(base_size=8)+
              geom_hline(aes(yintercept=-Inf)) + 
              geom_vline(aes(xintercept=-Inf)) + 
              coord_cartesian(clip="off")+
        #      coord_flip()+
         #     scale_x_discrete(limits=rev(c("WT.4w","Flt3.2w","Flt3.4w","Flt3.6w","Flt3.8w")))+
              theme(strip.text.y=element_text(angle=0),
                    strip.background = element_blank(),
                    axis.title.x = element_blank(),
                    axis.text.x=element_text(angle=30,hjust=1,vjust=1))


ggsave(
  plot_grid(ggSTAT_FTime,ggstem_FTime,ncol=1)
  ,width=2.5,height=6,
  file="./Mature_analysis/Final_Figures/Extended Data 2/ExData_2df.pdf")

```

Extended Data 2E
```{r}
Inflammation<- c("C3ar1","Ccl17","Ccl2","Ccl22","Ccl5","Cd40","Ciita",
                        "Csf2rb2","Cxcl10","Cxcl2","Fas","H2-DMa","H2-Eb1","Ido1",
                      "Irf4","Irf5","Irf7","Irf8","Irf9","Kynu","Mx1","Myd88",
                      "Nfkbia","Nfkbie","Nr4a1","Rel","Relb","Tnf","Stat1","Stat2")

long_counts_FTime%>%
  dplyr::filter(Gene%in%c(Inflammation))%>%
  dplyr::mutate(gene_set=ifelse(Gene%in%MAPK,"MAPK",
                         ifelse(Gene%in%Inflammation,"Inflammation", "Other")))%>%
  dplyr::mutate(Cell=factor(Cell,levels=c("LSK","GMP")))%>%
  dplyr::arrange(Cell,desc(Genotype),gene_set, Timepoint)%>%
  dplyr::mutate(value=scale(log2(value+0.1)))%>%
  tidy_heatmap(rows=Gene,
               columns = Sample,
               values = value,
               scale= "row",
               annotation_col =c(Cell,Genotype,Timepoint),
               cluster_rows=T,
               colors = rev(brewer.rdbu(n=100)),
               color_legend_n=100,
              # gaps_row = gene_set,
               treeheight_row = 0,
               cellwidth = 4,
               cellheight = 6,
               show_colnames = FALSE,
               annotation_colors = list("Genotype"=c("WT"="grey70",
                                                     "Flt3"=brewer.pal(n=5,"Reds")[5]),
                                        "Cell"=c("LSK"=brewer.pal(n=5,"Purples")[5],
                                                 "GMP"=brewer.pal(n=5,"Oranges")[5]),
                                          "Timepoint"=c("Control"=brewer.pal(n=5,"Greys")[1],
                                                      "2w"=brewer.pal(n=5,"Greys")[2],
                                                      "4w"=brewer.pal(n=5,"Greys")[3],
                                                      "6w"=brewer.pal(n=5,"Greys")[4],
                                                      "8w"=brewer.pal(n=5,"Greys")[5])) ,
               file="./Mature_analysis/Final_Figures/Extended Data 2/ExData_2e-heatmap.pdf")

```


Figure 1E
```{r}
library(tidyheatmaps)
genes_of_interest<- c("Gata1","Gata2","Socs1","Cish","Socs2",
                        "Itgax","Egr1","Junb","Fosb",'Jund',
                      "Tnf","Irf7","Irf8")
long_counts_FTime%>%
  dplyr::filter(Gene%in%c(genes_of_interest))%>%
  dplyr::arrange(Cell,desc(Genotype), Timepoint)%>%
  dplyr::mutate(value=scale(log2(value+0.1)))%>%
  dplyr::filter(Cell=="LSK")%>%
  tidy_heatmap(rows=Gene,
               columns = Sample,
               values = value,
               scale= "row",
               annotation_col =c(Cell,Genotype,Timepoint),
               cluster_rows=T,
               colors = rev(brewer.rdbu(n=100)),
               color_legend_n=100,
              # gaps_row = gene_set,
               treeheight_row = 0,
               cellwidth = 5,
               cellheight = 7,
               show_colnames = FALSE,
                legend = TRUE,

               annotation_colors = list("Genotype"=c("WT"="grey70",
                                                     "Flt3"=brewer.pal(n=5,"Reds")[5]),
                                        "Cell"=c("LSK"=brewer.pal(n=5,"Purples")[5],
                                                 "GMP"=brewer.pal(n=5,"Oranges")[5]),
                                          "Timepoint"=c("Control"=brewer.pal(n=5,"Greys")[1],
                                                      "2w"=brewer.pal(n=5,"Greys")[2],
                                                      "4w"=brewer.pal(n=5,"Greys")[3],
                                                      "6w"=brewer.pal(n=5,"Greys")[4],
                                                      "8w"=brewer.pal(n=5,"Greys")[5])) ,
               file="./Mature_analysis/Final_Figures/Figure 1/Figure1e-heatmap.pdf")

```



Figure 1F
GSEA setup
```{R gene set extraction and setup,cache=TRUE}
library(msigdbr)
library(fgsea)
C2_gene_sets <- msigdbr(species = "Mus musculus", category = "C2")%>%
                          data.frame%>%
                          dplyr::select(gene_symbol,gs_name)%>%
                          split(f = as.factor(.$gs_name))%>%
                          lapply(function(x){x$gene_symbol})
H_gene_sets <- msigdbr(species = "Mus musculus", category = "H")%>%
                          data.frame%>%
                          dplyr::select(gene_symbol,gs_name)%>%
                          mutate(gs_name=gsub("HALLMARK_","",gs_name))%>%
                          split(f = as.factor(.$gs_name))%>%
                          lapply(function(x){x$gene_symbol})


```

```{r}
library(DESeq2)
library(GSVA)
library(pals)
GMP_samples<-pheno_FTime%>%filter(Cell=="LSK")%>%pull(Sample)
vsd_transformation_FTime_GMP<-vsd_transformation_FTime[,GMP_samples]


vsd_FTime_scale<- (assay(vsd_transformation_FTime_GMP)-rowMeans(assay(vsd_transformation_FTime_GMP)))/rowSds(assay(vsd_transformation_FTime_GMP))
gsva_results_FTime<-gsva(vsd_FTime_scale,
                         H_gene_sets,
                      #    C2_gene_sets[grepl("STEM_CELL|MYELOID|FLT3",names(C2_gene_sets))],
                          min.sz=10, max.sz=5000,
                          method="ssgsea",
                         # rnaseq=TRUE, 
                          verbose=TRUE)%>%
                      data.frame()%>%
                      rownames_to_column(var="Gene_set")%>%
                      pivot_longer(cols=starts_with("IL"),names_to="Sample",values_to="value")%>%
                      left_join(pheno_FTime,by="Sample")%>%
                      unite(col=Composite,Genotype,Timepoint,sep = "_",remove = F)%>%
                      mutate(Composite=factor(Composite,levels=c("WT_4w","Flt3_2w","Flt3_4w","Flt3_6w","Flt3_8w")))
                    #  mutate(Genotype=factor(Genotype,levels=c("Flt3","WT")))

genes_of_interest<- grep("E2F|G2M|MITOTIC|INFLAMMAT",gsva_results_FTime$Gene_set,value = T)

ggssGSEA_GMP<-ggplot(gsva_results_FTime%>%
         # filter(Cell=="LSK")%>%
          filter(Gene_set%in%genes_of_interest)%>%
          mutate(Gene_set=factor(Gene_set)),
          aes(x=Composite, y=value,group=Gene_set,
           color=Gene_set))+
  stat_summary(fun.data = mean_se, width=0.25 , geom = "errorbar") + 
  stat_summary(fun=mean, geom="line", aes(group = Gene_set)) + 
  #facet_grid(.~Cell,scale="free")+
  ylab("Relative enrichment \n (ssGSEA score)")+
  theme_classic(base_size=8)+
  xlab("Group")+
  scale_color_manual(values=kelly(n=length(genes_of_interest)+1)[-1])+
 # guides(color="none")+
  #scale_y_continuous(limits=c(-0.2,1))+
   theme(strip.text.y=element_text(angle=0),
        axis.title.x = element_blank(),
        plot.title = element_blank(),
      #  legend.position = "right",
       # legend.title = element_blank(),
        axis.text.x=element_text(angle=30,hjust=1,vjust=1),
        strip.background = element_blank())

```

```{r}
library(DESeq2)
library(GSVA)
library(pals)
LSK_samples<-pheno_FTime%>%filter(Cell=="LSK")%>%pull(Sample)
vsd_transformation_FTime_LSK<-vsd_transformation_FTime[,LSK_samples]

vsd_FTime_scale<- (assay(vsd_transformation_FTime_LSK)-rowMeans(assay(vsd_transformation_FTime_LSK)))/rowSds(assay(vsd_transformation_FTime_LSK))
gsva_results_FTime<-gsva(vsd_FTime_scale,
                          C2_gene_sets[grepl("STEM_CELL|MYELOID|FLT3",names(C2_gene_sets))],
                          min.sz=10, max.sz=5000,
                          method="ssgsea",
                         # rnaseq=TRUE, 
                          verbose=TRUE)%>%
                      data.frame()%>%
                      rownames_to_column(var="Gene_set")%>%
                      pivot_longer(cols=starts_with("IL"),names_to="Sample",values_to="value")%>%
                      left_join(pheno_FTime,by="Sample")%>%
                      unite(col=Composite,Genotype,Timepoint,sep = "_",remove = F)%>%
                      mutate(Composite=factor(Composite,levels=c("WT_4w","Flt3_2w","Flt3_4w","Flt3_6w","Flt3_8w")))
                    #  mutate(Genotype=factor(Genotype,levels=c("Flt3","WT")))

genes_of_interest<- c(#"IL6_JAK_STAT3_SIGNALING",
                  #    "E2F_TARGETS",
                      "HEME_METABOLISM",
                  #   "KRAS_SIGNALING_DN",
                   "KRAS_SIGNALING_UP",
                     "IL2_STAT5_SIGNALING",
                   #   "TNFA_SIGNALING_VIA_NFKB",
                      "MYC_TARGETS_V2")

genes_of_interest<-c(#"WP_HEMATOPOIETIC_STEM_CELL_DIFFERENTIATION",
                     #"JAATINEN_HEMATOPOIETIC_STEM_CELL_UP",
                  #   "IVANOVA_HEMATOPOIESIS_STEM_CELL",   
"REACTOME_NEGATIVE_REGULATION_OF_FLT3",
"REACTOME_STAT5_ACTIVATION_DOWNSTREAM_OF_FLT3_ITD_MUTANTS",
  "BOQUEST_STEM_CELL_UP",
 "BROWN_MYELOID_CELL_DEVELOPMENT_UP"  ,
 "IVANOVA_HEMATOPOIESIS_STEM_CELL_LONG_TERM"        ,                   
"IVANOVA_HEMATOPOIESIS_STEM_CELL_SHORT_TERM"  )

ggssGSEA_LSK<-ggplot(gsva_results_FTime%>%
         # filter(Cell=="LSK")%>%
          filter(Gene_set%in%genes_of_interest)%>%
          mutate(Gene_set=factor(Gene_set)),
          aes(x=Composite, y=value,group=Gene_set,
           color=Gene_set))+
  stat_summary(fun.data = mean_se, width=0.25 , geom = "errorbar") + 
  stat_summary(fun=mean, geom="line", aes(group = Gene_set)) + 
  #facet_grid(.~Cell,scale="free")+
  ylab("Relative enrichment \n (ssGSEA score)")+
  theme_classic(base_size=8)+
  xlab("Group")+
  scale_color_manual(values=kelly(n=length(genes_of_interest)+1)[-1])+
 # guides(color="none")+
  #scale_y_continuous(limits=c(-0.2,1))+
   theme(strip.text.y=element_text(angle=0),
        axis.title.x = element_blank(),
        plot.title = element_blank(),
      #  legend.position = "right",
       # legend.title = element_blank(),
        axis.text.x=element_text(angle=30,hjust=1,vjust=1),
        strip.background = element_blank())

ggsave(ggssGSEA_LSK,#,textSize = 5),
       width=2,
       height=2,
       file="./Mature_analysis/Final_Figures/Figure 1/Figure1f-ssGSEA.pdf")
```

```{r}
library(tidyheatmaps)
library(tidyr)
library(broom)
significant_gene_sets_LSK<-gsva_results_FTime%>%
        #  dplyr::filter(Cell=="LSK")%>%
          group_by(Gene_set,Cell)%>%
          do(tidy(aov(value ~ Composite, data = .)))%>%
          dplyr::filter(term=="Composite")%>%
          arrange(p.value)%>%
          mutate(FDR=p.adjust(p=p.value,method="fdr",n = 50))%>%
          dplyr::filter(FDR<0.05)%>%
          pull(Gene_set)%>%unique()
          
gsva_results_FTime%>%
          dplyr::filter(Gene_set%in%significant_gene_sets_LSK)%>%
          group_by(Gene_set,Composite,Cell)%>%
          summarise(mean=mean(value))%>%
          mutate(mean_value=scale(mean))%>%
          ungroup()%>%
          unite(col=Composite_2,Composite,Cell,remove = FALSE)%>%
          arrange(Cell,Composite)%>%
          tidy_heatmap(rows=Gene_set,
                       columns=Composite_2,
                       values=mean,
                       gaps_col=Cell,
                       scale="row",
                       cluster_rows = TRUE,
                       cluster_cols = FALSE)
```

```{r}
gsva_results_FTime%>%
          group_by(Gene_set,Cell)%>%
          do(tidy(aov(value ~ Composite, data = .)))%>%
          dplyr::filter(term=="Composite")%>%
          arrange(p.value)%>%
          mutate(FDR=p.adjust(p=p.value,method="fdr",n = length(p.value)))%>%
          dplyr::select(Gene_set,Cell,Score=FDR)%>%
          tidyr::pivot_wider(names_from=Cell,values_from=Score)%>%
          ggplot(aes(x=-log10(GMP),y=-log10(LSK) ) )+
                  geom_point()#+scale_x_log10()+scale_y_log10()
         
```

```{r}
vsd_leuk_scale<- (assay(vsd_transformation_leuk)-rowMeans(assay(vsd_transformation_leuk)))/rowSds(assay(vsd_transformation_leuk))
gsva_results_leuk<-gsva(vsd_leuk_scale,
                          H_gene_sets,
                          min.sz=10, max.sz=5000,
                          method="ssgsea",
                         # rnaseq=TRUE, 
                          verbose=TRUE)%>%
                      data.frame()%>%
                      rownames_to_column(var="Gene_set")%>%
                      pivot_longer(cols=starts_with("IL"),names_to="Sample",values_to="value")%>%
                      left_join(pheno_select_leuk,by="Sample")%>%
                      unite(col=Composite,Genotype,Timepoint,sep = "_",remove = F)%>%
                      mutate(Composite=factor(Composite,levels=c("WT_4w","Flt3_4.5w","Npm1_4.5w","Flt3_Npm1_5w")))

genes_of_interest<- c(#"IL6_JAK_STAT3_SIGNALING",
                     #"E2F_TARGETS",
                    #  "HEME_METABOLISM",
                      "KRAS_SIGNALING_DN",
                    "KRAS_SIGNALING_UP")
                  #    "TNFA_SIGNALING_VIA_NFKB",
                   #   "MYC_TARGETS_V2")


ggplot(gsva_results_leuk%>%
         # filter(Cell=="LSK")%>%
          filter(Gene_set%in%genes_of_interest)%>%
          mutate(Gene_set=factor(Gene_set)),
          aes(x=Composite, y=value,group=Gene_set,
           color=Gene_set))+
  stat_summary(fun.data = mean_se, width=0.25 , geom = "errorbar") + 
  stat_summary(fun=mean, geom="line", aes(group = Gene_set)) + 
  facet_grid(.~Cell,scale="free")+
  ylab("Relative enrichment \n (ssGSEA score)")+
  theme_classic(base_size=14)+
  xlab("Group")+
  scale_color_manual(values=kelly(n=length(genes_of_interest)+1)[-1])+
  theme(strip.text.y=element_text(angle=0),
        axis.text.x=element_text(angle=30,hjust=1,vjust=1),
        strip.background = element_blank())

```

```{r}
library(tidyheatmaps)
library(tidyr)
library(broom)
significant_gene_sets_LSK<-gsva_results_leuk%>%
          dplyr::filter(Cell=="LSK")%>%
          group_by(Gene_set,Cell)%>%
          do(tidy(aov(value ~ Composite, data = .)))%>%
          dplyr::filter(term=="Composite")%>%
          arrange(p.value)%>%
          mutate(FDR=p.adjust(p=p.value,method="fdr",n = length(p.value)))%>%
          dplyr::filter(FDR<0.01)%>%
          pull(Gene_set)%>%unique()
          

gsva_results_leuk%>%
          dplyr::filter(Gene_set%in%significant_gene_sets_LSK)%>%
          group_by(Gene_set,Composite,Cell)%>%
          summarise(mean=median(value))%>%
         # mutate(mean_value=scale(mean))%>%
          ungroup()%>%
          unite(col=Composite_2,Composite,Cell,remove = FALSE)%>%
          arrange(Cell,Composite)%>%
          tidy_heatmap(rows=Gene_set,
                       columns=Composite_2,
                       values=mean,
                       gaps_col=Cell,
                       scale="row",
                       cluster_rows = TRUE,
                       cluster_cols = FALSE)
```

```{r LSk heatmap FLT3 Npm1}

genes_of_interest<-c ("Csda","Npm1","Cd34","Idh2","Cdk4","Shmt1","Myb","Gfi1","Mpo","Set","S100a8",'S100a9','Elane',"Myc",'Csf3r','Hoxa7',"Fos","Junb",'Egr1',"Flt3","Ets1","Cd74",'Jun','Socs2','Gata2',"Irf8","Dusp5","Il10ra","Csf2rb","Runx1","Itgax","Bcl2",'Ciita',"Mcl1","Fgfr1","Fosb",'Cish',"Tyrobp","Irf9","Irf4","Klf4","Il6st","Il6ra")

genes_of_interest<- c("Shmt2","Shmt1","Psat1","Phgdh","Slc1a4","Sfxn1","Mthfd2")

genes_of_interest<- c("Tal1","Meis1","Mecom","Hoxa9","Hoxb5",
                      "Cish","Socs2","Socs1",
                      "Hoxa7","Hoxa10","Flt3","Npm1","Cd34",
                      "Irf8","Irf7","Ccl5","Fgfr1",
                      "Mycn","Nr4a1")
 

results_Leuk$Flt3_Npm1_LSK$Flt3_Npm1_VS_Flt3%>%
                          data.frame()%>%
                          rownames_to_column(var="Gene")%>%
                          filter(padj<0.05&(log2FoldChange)<log2(2)*-1&baseMean>25)%>%
                          arrange(padj)
  

long_counts_leuk%>%
  filter(Gene%in%genes_of_interest)%>%
  filter(Cell%in%"GMP")%>%
  arrange(Cell,(Genotype), Timepoint)%>%
  tidy_heatmap(rows=Gene,
               columns = Sample,
               values = value,
               scale="row",
               annotation_col =c(Genotype),
               colors = coolwarm(n=100),
               
               treeheight_row = 0,
               color_legend_n=100,
               cellwidth = 5,
               cellheight = 5,
               show_colnames = FALSE,
               annotation_colors = list("Genotype"=c("WT"="grey70",
                                                     "Flt3"=brewer.pal(n=5,"Reds")[5],
                                                     "Npm1"=brewer.pal(n=5,"Blues")[5],
                                                     "Flt3_Npm1"=brewer.pal(n=5,"Greens")[5])#,
                                       # "Cell"=c("LSK"=brewer.pal(n=5,"Purples")[5],
                                              #   "GMP"=brewer.pal(n=5,"Oranges")[5])
                                       ),
                                         
               cluster_rows = TRUE)#,
              # file="./Results/2021/October/heatmap_of_Flt3_timecourey_GMP.pdf")


```

```{r}
results_Leuk$Flt3_Npm1_LSK$Flt3_Npm1_VS_Flt3%>%
                      data.frame()%>%
                      rownames_to_column(var="Gene")%>%
                      filter(Gene%in%c("Hoxa7","Hoxa9","Hoxa10"))
```


```{r}

ggplot(long_counts_leuk%>%
          filter(Gene%in%genes_of_interest)%>%
          mutate(Gene=factor(Gene,levels=genes_of_interest)),
          aes(x=Genotype,
                       y=value,
           fill=Genotype))+
  geom_boxplot()+#position = position_dodge(preserve = "single"))+
  facet_grid(Gene~Cell,scale="free")+
  ylab("Normalized expression")+
  theme_bw(base_size=14)+
  xlab("Group")+
  #scale_fill_manual(values=c("WT.4w"=brewer.pal(5,"Greys")[3],
  #                           "Flt3.2w"=brewer.pal(5,"Reds")[1],
  #                            "Flt3.4w"=brewer.pal(5,"Reds")[2],
  #                            "Flt3.6w"=brewer.pal(5,"Reds")[3],
   #                           "Flt3.8w"=brewer.pal(5,"Reds")[5]),"Group")+
  theme(strip.text.y=element_text(angle=0),
        axis.text.x=element_text(angle=30,hjust=1,vjust=1))

```